rotkehlchen/api/rest.py

```diff
@@ -4518,15 +4544,40 @@ def update_accounting_rule(
         except InputError as e:
             return api_response(wrap_in_fail_result(str(e)), status_code=HTTPStatus.CONFLICT)
 
+        self._invalidate_accounting_rule_cache(
+            event_type=event_type,
+            event_subtype=event_subtype,
+            counterparty=counterparty,
+        )
         return api_response(_wrap_in_ok_result(True), status_code=HTTPStatus.OK)
 
     def delete_accounting_rule(self, rule_id: int) -> Response:
         db = DBAccountingRules(self.rotkehlchen.data.db)
+        with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
```

Comment:
What are you doing here? Isn't this (and shouldn't this) logic be part of `db.remove_accounting_rule()`? It raises an `InputError` if this happens.

Reply:
I didn't do it because it was a extra query that might not be needed in other places but this function is used only here so I did as you suggested